### ISSUE TITLE
redirecting to Named pipe is not using ActionFileDefaultTemplate

### ISSUE THREAD

**Issue by rapadman** [rsyslog/rsyslog#373]
Recently migrated to RHEL 7.1 and in that, redirecting Named pipe is not using $ActionFileDefaultTemplate RSYSLOG_TraditionalFileFormat instead it uses FileFormat. However in earlier version 5.8.0 this was not the case, it was picking the configured format in $ActionFileDefaultTemplate. Is this an issue or any new configuration to be used for the same ?
Please suggest
rsyslogd 7.4.7, compiled with:        
FEATURE_REGEXP:                         Yes
        FEATURE_LARGEFILE:                      No
        GSSAPI Kerberos 5 support:              Yes
        FEATURE_DEBUG (debug build, slow code): No
        32bit Atomic operations supported:      Yes
        64bit Atomic operations supported:      Yes
        Runtime Instrumentation (slow code):    No
        uuid support:                           Yes

**Comment by rgerhards** [comment]
This is correct behaviour, because pipes are a different module. You can set a default for ompipe as follows:
```
module (path="builtin:ompipe" template="defaulTemplate")
```

**Comment by rapadman** [comment]
from /etc/rsyslog.conf
# Use default timestamp format
$ActionFileDefaultTemplate RSYSLOG_TraditionalFileFormat
module (path="builtin:ompipe" template="defaulTemplate")

rsyslogd -N1
rsyslogd: version 7.4.7, config validation run (level 1), master config /etc/rsyslog.conf
rsyslogd: error during parsing file /etc/rsyslog.conf, on or before line 32: parameter 'load' required but not specified - fix config [try http://www.rsyslog.com/e/2207 ]
rsyslogd: error during parsing file /etc/rsyslog.conf, on or before line 32: parameter 'template' not known -- typo in config file? [try http://www.rsyslog.com/e/2207 ]
rsyslogd: error during parsing file /etc/rsyslog.conf, on or before line 32: parameter 'path' not known -- typo in config file? [try http://www.rsyslog.com/e/2207 ]
rsyslogd: End of config validation run. Bye.

Above mentioned error observed, am I doing an mistake ? please suggest.

**Comment by rgerhards** [comment]
my fault. It looks liek the param is "load" instead of "path"

**Comment by rapadman** [comment]
Thanks for your fast response!
changed it to path. 
rsyslogd: ompipe: warning: default template was already set via legacy directive - may lead to inconsistent results. [try http://www.rsyslog.com/e/2220 ]
rsyslogd:  Could not find template 'defaulTemplate' - action disabled [try http://www.rsyslog.com/e/3003 ]
rsyslogd: error during parsing file /etc/rsyslog.conf, on or before line 96: errors occured in file '/etc/rsyslog.conf' around line 96 [try http://www.rsyslog.com/e/2207 ]
rsyslogd:  Could not find template 'defaulTemplate' - action disabled [try http://www.rsyslog.com/e/3003 ]
rsyslogd: error during parsing file /etc/rsyslog.conf, on or before line 97: errors occured in file '/etc/rsyslog.conf' around line 97 [try http://www.rsyslog.com/e/2207 ]
rsyslogd:  Could not find template 'defaulTemplate' - action disabled [try http://www.rsyslog.com/e/3003 ]
rsyslogd: error during parsing file /etc/rsyslog.conf, on or before line 98: errors occured in file '/etc/rsyslog.conf' around line 98 [try http://www.rsyslog.com/e/2207 ]
rsyslogd:  Could not find template 'defaulTemplate' - action disabled [try http://www.rsyslog.com/e/3003 ]
rsyslogd: error during parsing file /etc/rsyslog.conf, on or before line 101: errors occured in file '/etc/rsyslog.conf' around line 101 [try http://www.rsyslog.com/e/2207 ]

Changed to

# Use default timestamp format
$ActionFileDefaultTemplate RSYSLOG_TraditionalFileFormat

module (load="builtin:ompipe" template="RSYSLOG_TraditionalFileFormat")
[rsyslogd -N1
rsyslogd: version 7.4.7, config validation run (level 1), master config /etc/rsyslog.conf
rsyslogd: ompipe: warning: default template was already set via legacy directive - may lead to inconsistent results. [try http://www.rsyslog.com/e/2220 ]
rsyslogd: End of config validation run. Bye.

**Comment by Baerenbulle** [comment]
I have the same problem. I have to actions:
`$template ERXCorePipe,"/var/log/ER-X-Core/%SYSLOGFACILITY-TEXT%.log"
if ( $fromhost-ip=='192.168.65.1'  and $syslogfacility-text=="kern" ) then {
    action(type="omfile" DynaFile="ERXCorePipe")
    action(type="ompipe" Pipe="/var/log/ER-X-Core/kern.pipe.log")
}
`
In the log-File the date/time-Format is different compared to the pipe. 
`module (load="builtin:ompipe" template="RSYSLOG_TraditionalFileFormat")`
... throughs:
[rsyslogd -N1
rsyslogd: version 7.4.7, config validation run (level 1), master config /etc/rsyslog.conf
rsyslogd: ompipe: warning: default template was already set via legacy directive - may lead to inconsistent results.

**Comment by Baerenbulle** [comment]
Solved it:

`if ( $fromhost-ip=='192.168.65.1'  and $syslogfacility-text=="kern" ) then {`
`    action(type="omfile" DynaFile="ERXCorePipe")`
`    action(type="ompipe" Pipe="/var/log/ER-X-Core/kern.pipe.log" template="RSYSLOG_TraditionalFileFormat")`

**Comment by rgerhards** [comment]
Apologies for the very late reply. This is probably no longer relevant to the original reporter, but I am adding the conclusion for people who arrive here via search while we work through the backlog. We do not close rsyslog issues lightly, and we are not closing this merely because of age.

Triage result: valid non-code answer.

Reasoning: Named pipes use `ompipe`, not `omfile`, so `ActionFileDefaultTemplate` does not apply in the way expected. The working fix is to set the template on the `ompipe` action itself.

If this still reproduces on a current rsyslog release and maintained platform, please open a new issue with the current version, complete configuration, logs, and a minimal reproducer so we can investigate it as a fresh current problem.
